frame/contracts/src/wasm/prepare.rs

```diff
@@ -54,108 +56,50 @@ pub enum TryInstantiate {
-	/// Memory section contains declarations of internal linear memories, so if we find one
-	/// we reject such a module.
-	fn ensure_no_internal_memory(&self) -> Result<(), &'static str> {
-		if self.0.memory_section().map_or(false, |ms| ms.entries().len() > 0) {
-			return Err("module declares internal memory")
-		}
-		Ok(())
-	}
-
-	/// Ensures that tables declared in the module are not too big.
-	fn ensure_table_size_limit(&self, limit: u32) -> Result<(), &'static str> {
-		if let Some(table_section) = self.0.table_section() {
-			// In Wasm MVP spec, there may be at most one table declared. Double check this
-			// explicitly just in case the Wasm version changes.
-			if table_section.entries().len() > 1 {
-				return Err("multiple tables declared")
-			}
-			if let Some(table_type) = table_section.entries().first() {
-				// Check the table's initial size as there is no instruction or environment function
-				// capable of growing the table.
-				if table_type.limits().initial() > limit {
-					return Err("table exceeds maximum size allowed")
-				}
-			}
-		}
-		Ok(())
-	}
-
-	/// Ensure that any `br_table` instruction adheres to its immediate value limit.
-	fn ensure_br_table_size_limit(&self, limit: u32) -> Result<(), &'static str> {
-		let code_section = if let Some(type_section) = self.0.code_section() {
-			type_section
-		} else {
-			return Ok(())
-		};
-		for instr in code_section.bodies().iter().flat_map(|body| body.code().elements()) {
-			use self::elements::Instruction::BrTable;
-			if let BrTable(table) = instr {
-				if table.table.len() > limit as usize {
-					return Err("BrTable's immediate value is too big.")
-				}
-			}
-		}
-		Ok(())
-	}
-
-	fn ensure_global_variable_limit(&self, limit: u32) -> Result<(), &'static str> {
-		if let Some(global_section) = self.0.global_section() {
-			if global_section.entries().len() > limit as usize {
-				return Err("module declares too many globals")
-			}
-		}
-		Ok(())
-	}
-
-	fn ensure_local_variable_limit(&self, limit: u32) -> Result<(), &'static str> {
-		if let Some(code_section) = self.0.code_section() {
-			for func_body in code_section.bodies() {
-				let locals_count: u32 =
-					func_body.locals().iter().map(|val_type| val_type.count()).sum();
-				if locals_count > limit {
-					return Err("single function declares too many locals")
-				}
-			}
+	/// The inner Wasm module is checked not to have restricted WebAssembly proposals.
+	/// Returns `Err` if the `code` cannot be deserialized or if it contains an invalid module.
+	pub fn new<T>(
+		code: &[u8],
+		determinism: Determinism,
+		stack_limits: Option<StackLimits>,
+	) -> Result<Self, &'static str> {
+		// NOTE: wasmi does not support unstable WebAssembly features. The module is implicitly
+		// checked for not having those ones when creating `wasmi::Module` below.
+		let mut config = WasmiConfig::default();
+		config
+			.wasm_multi_value(false)
```

Comment:
some of them are stable features.

Reply:
Those that we set here, are stable. This comment means that those ones not listed below explicitly, they are unstable, and turned to `false` by default, i.e. implicitly.